src/xwayland.c

```diff
@@ -206,38 +206,25 @@ handle_set_class(struct wl_listener *listener, void *data)
 	view_update_app_id(view);
 }
 
-static void
-configure(struct view *view, struct wlr_box geo)
-{
-	if (!view->xwayland_surface) {
-		wlr_log(WLR_ERROR, "Not configuring view without xwayland_surface");
-		return;
-	}
-	view->pending_move_resize.update_x = geo.x != view->x;
-	view->pending_move_resize.update_y = geo.y != view->y;
-	view->pending_move_resize.x = geo.x;
-	view->pending_move_resize.y = geo.y;
-	view->pending_move_resize.width = geo.width;
-	view->pending_move_resize.height = geo.height;
-	wlr_xwayland_surface_configure(view->xwayland_surface, (int16_t)geo.x,
-	       (int16_t)geo.y, (uint16_t)geo.width, (uint16_t)geo.height);
-}
-
 static void
 move(struct view *view, double x, double y)
 {
+	assert(view && view->xwayland_surface);
```

Comment:
Please use two asserts() here. If it ever triggers we won't be able to tell the difference based on the log otherwise.

Reply:
I think I will just leave out the `view &&` part -- this is called as `view->impl->move(view, ...` so we'd crash before this point anyway if `view == NULL`.